dvc/rwlock.py

```diff
@@ -4,16 +4,20 @@
 from collections import defaultdict
 from contextlib import contextmanager
 
-from voluptuous import Schema, Invalid
+from voluptuous import Schema, Invalid, Optional, Extra
 from funcy.py3 import lmap, lfilter, lmapcat
 
 from .exceptions import DvcException
 from .lock import LockError
 from .utils import relpath
 
-INFO_SCHEMA = {"pid": int, "cmd": str}
-
-SCHEMA = Schema({"write": {str: INFO_SCHEMA}, "read": {str: [INFO_SCHEMA]}})
+SCHEMA = Schema(
+    {
+        Optional("write", default={}): {Extra: {"pid": int, "cmd": str}},
```

Comment:
IIRC, it is `Optional` by default. Or am I missing something?

Reply:
Not sure I understand why `Extra` is needed though. Could you elaborate?

Extra is used to define specific format of extra data that could be present without raising an exception. But I think you just need `{Required("pid"): int, Required("cmd"): str}` here

Reply:
@efiop , we have the `required=True` option, so it requires every key by default.

Without `Extra` it will require at least one object inside the `write` or `read`.


```python
>>> SCHEMA = Schema(
...     {
...         Optional("write", default={}): {str: {"pid": int, "cmd": str}},
...         Optional("read", default={}): {Extra: [{"pid": int, "cmd": str}]},
...     },
...     required=True,
... )

>>> SCHEMA({"write": {}, "read": {}})

MultipleInvalid: required key not provided @ data['write'][<class 'str'>]
```

Reply:
However, you are right, if we remove the `required=True` we only need to require the `INFO_SCHEMA`. I will make the change since it would be easier to review.